elements/lisk-p2p/src/peer_book/base_list.ts

```diff
@@ -13,146 +13,195 @@
-import { evictPeerRandomlyFromBucket, getBucketId, PEER_TYPE } from '../utils';
+import { P2PEnhancedPeerInfo, P2PPeerInfo } from '../p2p_types';
+import {
+	evictPeerRandomlyFromBucket,
+	getBucketId,
+	PEER_TYPE,
+	sanitizeEnhancedPeerInfo,
+} from '../utils';
 
 export interface PeerListConfig {
-	readonly peerBucketCount: number;
-	readonly peerBucketSize: number;
+	readonly numOfBuckets: number;
+	readonly bucketSize: number;
 	readonly secret: number;
 	readonly peerType: PEER_TYPE;
 }
 
-export interface CustomPeerInfo {
-	readonly peerInfo: P2PPeerInfo;
-	readonly dateAdded: Date;
+export type Bucket = Map<string, P2PEnhancedPeerInfo>;
+
+export interface BucketInfo {
+	readonly bucketId: number;
+	readonly bucket: Bucket;
 }
 
-// Base list class is covering a basic peer list that has all the functionality to handle buckets with default eviction strategy
 export class BaseList {
-	protected peerMap: Map<number, Map<string, CustomPeerInfo>>;
+	protected bucketIdToBucket: Map<number, Bucket>;
+	/* 
+		Auxilliary map for direct peerId => peerInfo lookups
+		Required because peerLists may be provided by discrete sources
+	*/
+	protected peerIdToPeerInfo: Map<string, P2PEnhancedPeerInfo>;
+	protected type: PEER_TYPE | undefined;
 	protected readonly peerListConfig: PeerListConfig;
 
 	public constructor({
-		peerBucketSize,
-		peerBucketCount,
+		bucketSize,
+		numOfBuckets,
 		secret,
 		peerType,
 	}: PeerListConfig) {
 		this.peerListConfig = {
-			peerBucketCount,
-			peerBucketSize,
+			bucketSize,
+			numOfBuckets,
 			peerType,
 			secret,
 		};
-		this.peerMap = new Map();
-		this.initPeerList(this.peerMap);
+		this.bucketIdToBucket = new Map();
+		this.initBuckets(this.bucketIdToBucket);
+		this.peerIdToPeerInfo = new Map();
 	}
 
-	public initPeerList(peerMap: Map<number, Map<string, CustomPeerInfo>>): void {
+	public initBuckets(bucketIdToBucket: Map<number, Bucket>): void {
 		// Init the Map with all the buckets
 		for (const bucketId of [
-			...new Array(this.peerListConfig.peerBucketCount).keys(),
+			...new Array(this.peerListConfig.numOfBuckets).keys(),
 		]) {
-			peerMap.set(bucketId, new Map<string, CustomPeerInfo>());
+			bucketIdToBucket.set(bucketId, new Map<string, P2PEnhancedPeerInfo>());
 		}
 	}
 
 	public get peerList(): ReadonlyArray<P2PPeerInfo> {
 		const peerListMap: P2PPeerInfo[] = [];
 
-		for (const peerMap of [...this.peerMap.values()]) {
-			for (const peer of [...peerMap.values()]) {
-				peerListMap.push(peer.peerInfo);
+		for (const peerList of [...this.bucketIdToBucket.values()]) {
+			for (const peer of [...peerList.values()]) {
+				// Remove internal fields before sharing
+				peerListMap.push(sanitizeEnhancedPeerInfo(peer));
 			}
 		}
 
 		return peerListMap;
 	}
 
-	public initPeerInfo = (peerInfo: P2PPeerInfo): CustomPeerInfo => ({
-		peerInfo,
-		dateAdded: new Date(),
-	});
-
-	public getBucket(ipAddress: string): Map<string, CustomPeerInfo> {
+	public calculateBucket(
```

Comment:
Please, organise the methods related to buckets altogether after the ones related to peers for example.